### PR TITLE
Add the upcoming releases & k8s min versions

### PR DESCRIPTION
**Pull request by dprotaso**
Related: https://github.com/knative/community/issues/245

Fixes 
- https://github.com/knative/docs/issues/2796 
- https://github.com/knative/docs/issues/2797

See the [rendered](https://github.com/knative/community/blob/c627bdfdb506a4e12f13db52cc006ec88b1ec242/mechanics/RELEASE-VERSIONING-PRINCIPLES.md) preview. I dropped older releases.

Based on the current principles we're going to make v1.17 our min K8s for the Knative release 0.20.x - but it will EOL 2-3 weeks later.

### CODE REVIEW

File: mechanics/RELEASE-VERSIONING-PRINCIPLES.md [hunk 1]
```diff
@@ -219,27 +219,23 @@ tuple
...
    </td>
    <td><strong>Availability</strong>
    </td>
   </tr>
-  <tr>
-   <td rowspan="3" >0.20.x
-   </td>
-   <td rowspan="3" >2020-12-22
+    <tr>
+   <td>0.21.x
```
Comment (mattmoor) [inline mechanics/RELEASE-VERSIONING-PRINCIPLES.md:230]: Can't comment on it but I noticed `LIfe` has weird capitalization.

File: mechanics/RELEASE-VERSIONING-PRINCIPLES.md [hunk 2]
```diff
@@ -609,276 +546,42 @@ tuple
...
-   <td>Lemonade
-   </td>
-   <td>default 
-<p>
-(deprecated)
-   </td>
-  </tr>
-  <tr>
-   <td rowspan="3" >0.9.x
+   <td rowspan="2" >0.20.x
```
Comment (vaikas) [inline mechanics/RELEASE-VERSIONING-PRINCIPLES.md:550]: Just adding it here, but for Eventing, v1beta1 can be left as optional (deprecated) and maybe in .20 we'll not serve it, but currently it's still being served.

**Comment by dprotaso** [comment]
/hold cancel

Should be good now

**Pull request merged**